features/generator_features.py

```diff
@@ -171,15 +171,28 @@ class IncrementalRotation(traits.HasTraits):
     delta_rotation_x = traits.Float(0.0, desc="rotation step size about bmi3d x-axis in degrees")
 
     trials_per_increment = traits.Int(1, desc="number of successful trials per rotation step")
+    final_tracking_out_time = traits.Float(1.6, desc="Time allowed to be tracking outside the target for final rotation") # AKA tolerance time
 
     def init(self):    
         super().init()
         self.num_trials_success = 0
-        self.num_increments = int( (self.final_rotation_y-self.init_rotation_y) / self.delta_rotation_y+1 )
+        # NOTE: currently can only handle incremental rotations around one axis at a time
+        if self.final_rotation_y != self.init_rotation_y:
+            rotation_info = [self.init_rotation_y, self.final_rotation_y, self.delta_rotation_y]
+        elif self.final_rotation_z != self.init_rotation_z:
+            rotation_info = [self.init_rotation_z, self.final_rotation_z, self.delta_rotation_z]
+        elif self.final_rotation_x != self.init_rotation_x:
+            rotation_info = [self.init_rotation_x, self.final_rotation_x, self.delta_rotation_x]
+        else:
+            print('Warning! Please specify an initial and final rotation.')
+
+        self.num_increments = int( (rotation_info[1]-rotation_info[0]) / rotation_info[2]+1 )
```

Comment:
it would be wise to fix this even if it's simple, e.g.
```suggestion
        # NOTE: currently can only handle incremental rotations around one axis at a time
        if self.final_rotation_y != self.init_rotation_y:
            num_increments_y = int( (self.final_rotation_y-self.init_rotation_y) / self.delta_rotation_y+1 )
        elif self.final_rotation_z != self.init_rotation_z:
            num_increments_z = int( (self.final_rotation_z-self.init_rotation_z) / self.delta_rotation_z+1 )
        elif self.final_rotation_x != self.init_rotation_x:
            num_increments_x = int( (self.final_rotation_x-self.init_rotation_x) / self.delta_rotation_x+1 )
        else:
            print('Warning! Please specify an initial and final rotation.')

        self.num_increments = np.max([num_increments_y, num_increments_z, num_increments_x])
```

and then check that you haven't exceeded the final_rotation_xyz for each inside _start_wait()